qiskit_ibm_provider/transpiler/passes/scheduling/scheduler.py

```diff
@@ -0,0 +1,274 @@
+        self._current_block_measures = set()
+        self._bit_indices = {q: index for index, q in enumerate(dag.qubits)}
+
+    def _get_duration(self, node: DAGNode) -> int:
+        return super()._get_node_duration(node, self._bit_indices, self._dag)
+
+    def _visit_node(self, node: DAGNode) -> None:
+        # compute t0, t1: instruction interval, note that
+        # t0: start time of instruction
+        # t1: end time of instruction
+        if isinstance(node.op, self.CONDITIONAL_SUPPORTED):
+            self._visit_conditional_node(node)
+        else:
+            if node.op.condition_bits:
+                raise TranspilerError(
+                    f"Conditional instruction {node.op.name} is not supported in ASAP scheduler."
+                )
+
+            if isinstance(node.op, Measure):
+                self._visit_measure(node)
+            elif isinstance(node.op, Reset):
+                self._visit_reset(node)
+            else:
+                self._visit_generic(node)
+
+    def _visit_conditional_node(self, node: DAGNode) -> None:
+        """Handling case of a conditional execution.
+
+        Conditional execution durations are currently non-deterministic. as we do not know
+        the time it will take to begin executing the block. We do however know the
+        duration of the block contents execution (provided it does not also contain
+        conditional executions).
+
+        TODO: Update for support of general control-flow, not just single conditional operations.
+        """
+        # Special processing required to resolve conditional scheduling dependencies
+        if node.op.condition_bits:
+            # Trigger the start of a conditional block
+            self._begin_new_circuit_block()
+
+            op_duration = self._get_duration(node)
+
+            t0q = max(self._idle_after[q][1] for q in node.qargs)
+            # conditional is bit tricky due to conditional_latency
+            t0c = max(self._idle_after[bit][1] for bit in node.op.condition_bits)
+            if t0q > t0c:
+                # This is situation something like below
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒░░
+                # C ▒▒▒░░░░░░░░
+                #     |t0c
+                #
+                # In this case, you can insert readout access before tq0
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒▒▒
+                # C ▒▒▒░░░▒▒░░░
+                #         |t0c
+                #
+                t0c = t0q
+            t1c = t0c
+            for bit in node.op.condition_bits:
+                # Lock clbit until state is read
+                self._idle_after[bit] = (self._current_block_idx, t1c)
+
+            # It starts after register read access
+            t0 = max(t0q, t1c)  # pylint: disable=invalid-name
+
+            t1 = t0 + op_duration  # pylint: disable=invalid-name
+            self._update_idles(node, t0, t1)
```

Comment:
Just to confirm; in the current backend classical registers are also blocked until qubit operation completes? I think this code should be fine anyways because there is no instruction that can be conditional and takes both quantum and classical registers.